java/src/main/native/include/jni_utils.hpp

```diff
@@ -752,14 +767,14 @@ inline jthrowable cuda_exception(JNIEnv *const env, cudaError_t status, jthrowab
     return NULL;
   }
 
-  jobject ret = env->NewObject(ex_class, ctor_id, msg, cause);
+  jint err_code = static_cast<jint>(status);
+
+  jobject ret = env->NewObject(ex_class, ctor_id, msg, err_code, cause);
   return (jthrowable)ret;
 }
 
 inline void jni_cuda_check(JNIEnv *const env, cudaError_t cuda_status) {
   if (cudaSuccess != cuda_status) {
-    // Clear the last error so it does not propagate.
-    cudaGetLastError();
     jthrowable jt = cuda_exception(env, cuda_status);
     if (jt != NULL) {
```

Comment:
Not an issue created by this PR, but just noticed this code is not quite correct.  If `jt == NULL` we cannot throw a JVM exception, but that means a different JVM exception will be pending (e.g.: OOM error or method not found error) due to the error that occurred while trying to create the exception object.  We always want to throw the C++ exception here, not just when the `jthrowable` is not null.

Reply:
Added a throw clause at the else branch as well.